### PR TITLE
Users: clean up OrgId when all user orgs are removed

### LINKED ISSUE #20224: [BUG] User removed from organization is still marked as in it
**What happened**:
If you remove a user from all organizations in the system, the call to `/api/users/:id` will still return the field `orgId` as its last known organization.

**What you expected to happen**:
Return `null` in the field.

**How to reproduce it (as minimally and precisely as possible)**:
1. Create a user
2. Remove it from all organizations (probably via the API)
3. Call `/api/users/:id`

**Environment**:
- Grafana version: 6.4.3
- OS Grafana is installed on: Linux
- Others: Postgres Database

### PR DESCRIPTION
**Pull request by ifrost**
**What this PR does / why we need it**:

This is to fix #20224. With the fix, when all user orgs are removed we now set `OrgId` to `0` and omit it from the `/api/users/:id` response.

**Which issue(s) this PR fixes**:

Fixes #20224

**Special notes for your reviewer**:

How to test?

Add a new user and check the user id:

Screen Shot 2022-03-01 at 12 36 11

Go to `http://localhost:3000/api/users` + `user id` (e.g. http://localhost:3000/api/users/4). The response should contain automatically assigned `orgId`:

```json
{
  "id": 4,
  "email": "user@user.user",
  ...
  "orgId": 1,
}
```

Remove the user from the organization and check the response from `/api/users/...`. It should not contain `orgId` any more.

### CODE REVIEW

**Comment by ifrost** [comment]
@grafana/backend-platform could someone please have a look a this?

**Review by papagian — CHANGES_REQUESTED** [review]
Thank you for your contribution! LGTM, I only have a minor suggestion.

Also, regarding [this observed behaviour](https://github.com/grafana/grafana/issues/20224#issuecomment-579867298), when we fetch the signed in user from the database [we explicitly set the org to `-1` if it's missing](https://github.com/grafana/grafana/blob/c15ddc295cc10560bb94fe4567a9b33f61ab8496/pkg/services/sqlstore/user.go#L386) so I assume that it's intentional. In any case, I have tested removing a user with admin privileges from an org and after removal the user is able to login but with no privileges.

File: pkg/services/sqlstore/user.go [hunk 1]
```diff
@@ -470,7 +470,7 @@ func setUsingOrgInTransaction(sess *DBSession, userID int64, orgID int64) error
 		OrgId: orgID,
 	}
 
-	_, err := sess.ID(userID).Update(&user)
+	_, err := sess.ID(userID).MustCols("org_id").Update(&user)
```
[the remark this answers is not in the document]
Comment (ifrost) [reply pkg/services/sqlstore/user.go:454]: To make it more explicit I've created a separate function for removing an org in 489c4f36273adc488e317284d50b394a1d62e247. Would that make sense? Or should I just go with an extra bool param?

[the remark this answers is not in the document]
Comment (papagian) [reply pkg/services/sqlstore/user.go:454]: that's even better, thank you!

**Review by papagian — APPROVED** [review]

**Pull request merged**